<launch>
    <include file="$(find turtlebot3_bringup)/launch/turtlebot3_core.launch" />
    <include file="$(find velodyne_pointcloud)/launch/VLP16_points.launch" />

    <machine name="pi" address="jalapeno.local" env-loader="~/catkin_ws/devel/env_pi.sh" user="aa274" password="aa274" />



    <node machine="pi" type="raspicam_node" pkg="raspicam_node" name="raspicam_node" output="screen">

        <param name="camera_info_url" value="package://raspicam_node/camera_info/camerav2_410x308.yaml"/>
        <param name="width" value="410"/>
        <param name="height" value="308"/>

        <param name="framerate" value="30"/>
        <param name="exposure_mode" value="antishake"/>
        <param name="shutter_speed" value="0"/>

        <param name="camera_frame_id" value="raspicam"/>
    </node>

    <node pkg="AA274_Final_Project" type="camera_relay.py" name="camera_relay"/>

    <node pkg="gmapping" type="slam_gmapping" name="turtlebot3_slam_gmapping" output="screen">
        <param name="base_frame" value="base_footprint"/>
        <param name="odom_frame" value="odom"/>
        <param name="map_update_interval" value="2.0"/>
        <param name="maxUrange" value="4.0"/>
        <param name="minimumScore" value="100"/>
        <param name="linearUpdate" value="0.2"/>
        <param name="angularUpdate" value="0.2"/>
        <param name="temporalUpdate" value="0.5"/>
        <param name="delta" value="0.05"/>
        <param name="lskip" value="0"/>
        <param name="particles" value="120"/>
        <param name="sigma" value="0.05"/>
        <param name="kernelSize" value="1"/>
        <param name="lstep" value="0.05"/>
        <param name="astep" value="0.05"/>
        <param name="iterations" value="5"/>
        <param name="lsigma" value="0.075"/>
        <param name="ogain" value="3.0"/>
        <param name="srr" value="0.01"/>
        <param name="srt" value="0.02"/>
        <param name="str" value="0.01"/>
        <param name="stt" value="0.02"/>
        <param name="resampleThreshold" value="0.5"/>
        <param name="xmin" value="-10.0"/>
        <param name="ymin" value="-10.0"/>
        <param name="xmax" value="10.0"/>
        <param name="ymax" value="10.0"/>
        <param name="llsamplerange" value="0.01"/>
        <param name="llsamplestep" value="0.01"/>
        <param name="lasamplerange" value="0.005"/>
        <param name="lasamplestep" value="0.005"/>
    </node>


  <!-- TO DO: FIX CONTROLLER/PARAMS -->
  <rosparam param="sim">false</rosparam>
  <rosparam param="map">true</rosparam>
  <rosparam param="rviz">true</rosparam>
  <rosparam param="use_tf">false</rosparam>
  <node pkg="asl_turtlebot" type="supervisor_nav.py" name="turtlebot_supervisor_nav" output="screen"/>
  <node pkg="asl_turtlebot" type="detector.py" name="turtlebot_detector" output="screen"/>
  <node pkg="asl_turtlebot" type="pose_controller_nav.py" name="turtlebot_pose_controller_nav" />
  <node pkg="asl_turtlebot" type="navigator.py" name="turtlebot_navigator" />
  
  <!-- Lane Detection -->
  <!-- <node pkg="asl_turtlebot" type="detect_lane_node.py" name="detect_lane_node" /> -->

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find asl_turtlebot)/rviz/nav.rviz"/>

  <!-- <node pkg="tf" type="static_transform_publisher" name="camera_true_broadcaster" args="0.03 0 0.067 0.707107 0 0.707107 0 base_footprint raspicam 100" />

  <node pkg="tf" type="static_transform_publisher" name="camera_broadcaster" args="0 0 0 0 0 0.707107 0.707107 raspicam camera 100" />

  <node pkg="tf" type="static_transform_publisher" name="velodyne_broadcaster" args="-0.04 0 0.215 0 0 0 1 base_footprint velodyne 100" /> -->
</launch>
